vllm/attention/backends/torch_sdpa.py

```diff
@@ -182,30 +200,66 @@ def forward(
                             attn_metadata.seq_lens, self.sliding_window,
                             query.dtype)  # type: ignore
                     else:
-                        att_masks = [None] * len(attn_metadata.seq_lens)
+                        if self.fuse_batch:
+                            att_masks = _make_sliding_window_bias(
+                                attn_metadata.seq_lens,
+                                None,
+                                dtype=query.dtype)
+                        else:
+                            att_masks = [None] * len(attn_metadata.seq_lens)
                     attn_metadata.attn_bias = att_masks
 
-                query = query.movedim(0, query.dim() - 2)
-                key = key.movedim(0, key.dim() - 2)
-                value = value.movedim(0, value.dim() - 2)
-
-                start = 0
-                output = torch.empty(
-                    (num_tokens, self.num_heads, self.head_size),
-                    dtype=query.dtype)
-                for seq_len, mask in zip(attn_metadata.seq_lens,
-                                         attn_metadata.attn_bias):
-                    end = start + seq_len
-                    sub_out = scaled_dot_product_attention(
-                        query[:, start:end, :],
-                        key[:, start:end, :],
-                        value[:, start:end, :],
-                        attn_mask=mask,
-                        dropout_p=0.0,
-                        is_causal=not self.need_mask,
-                        scale=self.scale).movedim(query.dim() - 2, 0)
-                    output[start:end, :, :] = sub_out
-                    start = end
+                if self.fuse_batch:
+                    out = torch.empty(
+                        (num_tokens, self.num_heads, self.head_size),
+                        dtype=query.dtype,
+                        device=query.device)
+                    tmp = [0]
+                    tmp.extend(attn_metadata.seq_lens)
+                    seqlen = torch.tensor(tmp)
+                    max_seqlen = max(attn_metadata.seq_lens)
+                    seqlen_q = torch.cumsum(seqlen,
+                                            dim=0).to(device=query.device)
+                    from vllm._ipex_ops import ipex_ops
```

Comment:
Could you add a comment that the lazy import is required because the code is shared with the CPU backend?

Reply:
I have create a new `IpexAttnBackend` just like tpu side do(due to paged attention kernels are not compiled), xpu path will no longer use this backend, And the new backend will import ipex by default.